server/athenian/api/internal/features/entries.py

```diff
@@ -1861,156 +2061,9 @@ async def _calc_pull_request_facts_github(
-    return ";".join(
-        ",".join("%s:%s" % (k.name, sorted(v)) for k, v in sorted(p.items())) for p in participants
-    )
-
-
-def _intersect_items_groups(
-    n_groups: int,
-    n_items: int,
-    *group_arrays: np.ndarray,
-) -> list[np.ndarray]:
-    """Intersect lists of groups.
-
-    Each group in the result will be the intersection of groups in `group_arrays` index by index.
-    Each group_arrays must generate `n_groups`.
-
-    """
-    groups = np.empty(n_groups, dtype=object)
-    group = np.empty(n_items, dtype=np.uint8)
-    for i, group_masks in enumerate(zip(*group_arrays)):
-        group[:] = 0
-        for mask in group_masks:
-            group[mask] += 1
-        groups[i] = np.flatnonzero(group == len(group_masks))
-    return groups
-
-
-def make_calculator(
-    account: int,
-    meta_ids: tuple[int, ...],
-    mdb: Database,
-    pdb: Database,
-    rdb: Database,
-    cache: Optional[aiomcache.Client],
-) -> MetricEntriesCalculator:
-    """Get the metrics calculator according to the account's features."""
-    return MetricEntriesCalculator(
-        account, meta_ids, DEFAULT_QUANTILE_STRIDE, mdb, pdb, rdb, cache,
-    )
-
-
-@dataclasses.dataclass(frozen=True, slots=True)
-class MetricsLineRequest:
-    """Common base for multiple metrics request classes."""
-
-    metrics: Sequence[str]
-    time_intervals: Sequence[Sequence[datetime]]
-    teams: Sequence[TeamSpecificFilters]
-    metric_params: dict[str, Any] = dataclasses.field(default_factory=dict)
-
-    def __str__(self) -> str:
-        """Summarize the request."""
-        return (
-            f"<[{', '.join(self.metrics)}],"
-            f" [{', '.join('[%s]' % ', '.join(str(d) for d in s) for s in self.time_intervals)}],"
-            f" [{', '.join(str(f) for f in self.teams)}]>"
-        )
-
-    def __sentry_repr__(self) -> str:
-        """Format for Sentry reports."""
-        return str(self)
-
-    def all_jira_filters(self) -> Iterator[JIRAFilter]:
-        """Collect all JIRAFilter from the included TeamSpecificFilters."""
-        return (td.jira_filter for td in self.teams)
-
-
-@dataclasses.dataclass(frozen=True, slots=True)
-class TeamSpecificFilters:
-    """Filters that are different for each team."""
-
-    team_id: int
-    # filters start here
-    repositories: tuple[str]
-    participants: PRParticipants | ReleaseParticipants | JIRAParticipants
-    jira_filter: JIRAFilter
-    # add more filters here
-
-    def __str__(self) -> str:
-        """Format the filters as a stable string."""
-        # participants are fully defined by the team ID
-        return f"{self.team_id}|{','.join(self.repositories)}|{self.jira_filter}"
-
-    def __sentry_repr__(self):
-        """Format for Sentry."""
-        dikt = dataclasses.asdict(self)
-        del dikt["participants"]  # fully defined by the team ID
-        return str(dikt)
+def _merge_pr_participants(participants: Iterable[PRParticipants]) -> PRParticipants:
```

Comment:
Can you please move this function under some class, so that we don't pollute the global scope.

Reply:
moved to a new class
https://github.com/athenianco/athenian-api/pull/3039/commits/faec3c7653d9c9c0fdff06b27eb543ce06c7dba8